Arrowgene.Ddon.GameServer/Handler/InstanceEnemyKillHandler.cs

```diff
@@ -173,7 +174,19 @@ public override S2CInstanceEnemyKillRes Handle(GameClient client, C2SInstanceEne
                     }
                 }
 
-                uint calcExp = _gameServer.ExpManager.GetAdjustedExp(client.GameMode, RewardSource.Enemy, client.Party, enemyKilled.GetDroppedExperience(), enemyKilled.Lv);
+                uint baseEnemyExp = 0;
+                if (Server.GameLogicSettings.Get<bool>("GameLogicSettings", "EnableExpCalculationMixin"))
+                {
+                    var expCurveMixin = Server.ScriptManager.MixinModule.Get<IExpCurveMixin>("exp_curve");
+                    baseEnemyExp = expCurveMixin.GetExpValue(enemyKilled);
+                }
+                else
+                {
+                    baseEnemyExp = enemyKilled.GetDroppedExperience();
```

Comment:
Indeed i have opinions. We could take this chance to make the exp value in the json nullable. If its null, use the calculated value, if its not, use the json value. A setting could toggle whether these custom values are used or not 🤔

Reply:
This is going to make exp previews in DDOn Tools a total nightmare to deal with

Reply:
How hard what it be to have a way to configure default exp calculation by the user? Does Godot have an evaluate feature where we can provide some code in an input box or something in the settings?

Reply:
I'm almost sure it's possible to do, but the problem is that you'd have to port over whatever your server has set up to Godot's script for DDOn Tools 🤔

Reply:
For this PR, I am going to leave in the hook, but I am setting it to just have the same behavior as the default exp scheme derived from the EnemySpawn.json. Not sure if ddon-tools has some way to identify that something is "nullable" so the parser can support that. Seems like a task left for a focused PR instead of this more general one.

Reply:
null is standard javascript so im sure godot's JSON parser can handle it. if we're doing that later then this PR is ready to go